plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-product-variations-v2-controller.php

```diff
@@ -137,10 +137,48 @@ public function register_routes() {
 	 *
 	 * @since  3.0.0
 	 * @param  int $id Object ID.
-	 * @return WC_Data
+	 * @return WC_Data|null
 	 */
 	protected function get_object( $id ) {
-		return wc_get_product( $id );
+		$object = wc_get_product( $id );
+		return ( $object && 0 !== $object->get_parent_id() ) ? $object : null;
+	}
+
+	/**
+	 * Checks that a variation belongs to the specified parent product.
+	 *
+	 * @param int $variation_id Variation ID.
+	 * @param int $parent_id    Parent product ID to check against.
+	 * @return bool TRUE if variation and parent product exist. FALSE otherwise.
+	 *
+	 * @since 9.2.0
+	 */
+	protected function check_variation_parent( int $variation_id, int $parent_id ) {
```

Comment:
Optional, but we could also specify the return type:

```suggestion
	protected function check_variation_parent( int $variation_id, int $parent_id ): bool {
```